lib/dash/content_protection.js

```diff
@@ -262,6 +262,169 @@ shaka.dash.ContentProtection.parseFromRepresentation = function(
 };
 
 
+/**
+ * Gets a Widevine license URL from a content protection element
+ * containing a custom `ms:laurl` element
+ *
+ * @param {shaka.dash.ContentProtection.Element} element
+ * @return {string}
+ */
+shaka.dash.ContentProtection.getWidevineLicenseUrl = function(element) {
+  const mslaurlNode = shaka.util.XmlUtils.findChild(element.node, 'ms:laurl');
+  if (mslaurlNode) {
+    return mslaurlNode.getAttribute('licenseUrl') || '';
+  }
+  return '';
+};
+
+
+/**
+* @typedef {{
+*   type: number,
+*   value: string
+* }}
+*
+* @description
+* The parsed result of a PlayReady object record.
+*
+* @property {number} type
+*   Type of data stored in the record.
+* @property {string} value
+*   Record content.
+*/
+shaka.dash.ContentProtection.PlayReadyRecord;
+
+/**
+ * Enum for PlayReady record types.
+ * @enum {number}
+ */
+shaka.dash.ContentProtection.PLAYREADY_RECORD_TYPES = {
+  RIGHTS_MANAGEMENT: 0x001,
+  RESERVED: 0x002,
+  EMBEDDED_LICENSE: 0x003,
+};
+
+
+/**
+* Parses a Uint16Array for PlayReady Object Records.
+* Each PRO Record is preceeded by it's PlayReady
+* Record type and length in bytes.
+*
+* PlayReady Object Record format: https://goo.gl/FTcu46
+*
+* @param {Uint16Array} recordData
+* @return {!Array.<shaka.dash.ContentProtection.PlayReadyRecord>}
+* @private
+*/
+shaka.dash.ContentProtection.parseMsProRecords_ = function(
+  recordData) {
+  let head = 0;
+  let records = [];
+
+  while (head < recordData.length) {
+    const recordRaw = recordData.subarray(head);
+    if (recordRaw.length > 1) {
+      let offset = 0;
+      const type = recordRaw[offset++];
+      const byteLength = recordRaw[offset++];
+      // Convert from bytes to WORD length
+      const wordLength = byteLength / 2;
```

Comment:
Is the byte length required to be even, or else the division would fail, right? Could we add a `goog.asserts.assert` call there so that we can be sure that we are parsing correctly?

Reply:
Added.